web/api/v1/api_test.go

```diff
@@ -1207,6 +1250,56 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, es storage.E
 				},
 			},
 		},
+		{
+			endpoint: api.query,
+			query: url.Values{
+				"query": []string{
+					`label_replace(vector(42), "foo", "bar", "", "") or label_replace(vector(3.1415), "dings", "bums", "", "")`,
+				},
+				"time": []string{"123.4"},
```

Comment:
I would add a `"limit": []string{"2"},` here and then repeat the test with `"limit": []string{"1"},` and `"limit": []string{"0"},`.